docs/source/1.0/spec/core/behavior-traits.rst

```diff
@@ -655,3 +655,228 @@ See
             input: PutSomethingInput,
             output: PutSomethingOutput
         }
+
+
+.. _httpChecksum-trait:
+
+----------------------
+``httpChecksum`` trait
+----------------------
+
+Summary
+    Indicates that an operation's HTTP request or response supports checksum
+    validation. At least one of request or response checksum properties must
+    be specified within the trait.
+
+Trait selector
+    ``operation``
+Value type
+    ``structure``
+
+The ``httpChecksum`` trait is a structure that contains the following members:
+
+.. list-table::
+    :header-rows: 1
+    :widths: 10 10 80
+
+    * - Property
+      - Type
+      - Description
+    * - request
+      - :ref:`HttpChecksumProperties structure <checksum-properties>`
+      - The ``request`` property defines checksum validation behavior for
+        HTTP requests.
+
+    * - response
+      - :ref:`HttpChecksumProperties structure <checksum-properties>`
+      - The ``response`` property defines checksum validation behavior for
+        HTTP responses.
+
+.. tabs::
+
+    .. code-tab:: smithy
+
+        @httpChecksum(
```

Comment:
Did we consider making these a little more explicit, customizable, and future proof by using a map similar to:

```
@httpChecksum(
    request: {
        sha256: {
            header: "x-checksum-sha256"
        },
        crc32: {
            header: "x-checksum-crc32"
        }
    }
)
```

Here, `request` and `response` are maps where each key is the algorithm, and each value is a union of possible locations (i.e., it can be set to `header` or `trailer`, and we could expand to `query` in the future for requests if needed). The value of the union is the binding location name (i.e., header name or trailing header name).

This provides more flexibility than the current proposal because each algorithm can use whatever header or query string parameter is necessary. The header/query binding is explicit and does not need prefixes. The names can now support anything, including `Content-MD5` if we ever wanted to support it (we probably don't but just an example).

Implementations don't need to perform any prefix logic to determine the header name either.

Reply:
I like the idea, have few concerns. Let's sync offline on this.